---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning =  FALSE,
  message = FALSE
)
library(knitr)
library(kableExtra)
```

# cinaR
`cinaR` is an end-to-end computational pipeline for bulk ATAC-seq profiles. 
It starts from consensus peak files and outputs the Differentially Accessible (DA)
peaks.

## Installation

```{r eval=FALSE}
library(devtools)
install_github("eonurk/cinaR")
```


## Quick Start

```{r}
library(cinaR)
data("atac_seq_consensus_bm")
```

Bed formatted consensus matrix (chr, start, end and samples)
```{r}
dim(bed)
```


```{r}
# bed formatted file
head(bed[,1:5])
```

Create the contrasts you want to compare, here we create contrasts for 
22 mice samples from different strains.
```{r}
# create contrast vector which will be compared.
contrasts <- sapply(strsplit(colnames(bed), split = "-", fixed = T), 
                    function(x){x[1]})[4:25]
contrasts
```

`cinaR` function directly computes the differentially accessible peaks.
```{r cache=TRUE}
# If reference genome is not set hg38 will be used!
results <- cinaR(bed, contrasts, reference.genome = "mm10")
```

```{r}
colnames(results[[1]])
```


```{r }
head(results[[1]][,1:5])
```



## Creating different contrasts

Note that you can further divide the resolution of contrasts, for instance this
is also a valid vector
```{r}
contrasts <- sapply(strsplit(colnames(bed), split = "-", fixed = T), 
                    function(x){paste(x[1:4], collapse = "-")})[4:25]
unique(contrasts)
```
in this case, each of them will be compared to each other which will result in
28 different comparisons.

## Saving DA peaks to excel

Setting `save.DA.peaks = TRUE` in `cinaR` function will create a `DApeaks.xlsx` file
in the current directory. This file includes all the comparisons in different tabs.
Additionally, you can set the path/name of the file using 
`DA.peaks.path` argument after setting `save.DA.peaks = TRUE`.

For instance, 

```{r eval=FALSE}
results <- cinaR(bed, contrasts, reference.genome = "mm10", 
                 save.DA.peaks = T, DA.peaks.path = "./Peaks_mice.xlsx")
```

will create an excel file with name `Peaks_mice.xlsx` in the current directory.

## Using different GLM algorithms
Currently, `cinaR` supports 4 different algorithms, namely; 

- edgeR
- limma-voom
- limma-trend
- DESeq2

You can change the used algorithm by simply setting `DA.choice` argument.
For more information, `?cinaR`


## Contribution

You can send pull requests to make your contributions.

I occasionally mess up, so all comments are appreciated!

## Future work
- Add enrichment pipeline
  - hyper-geometric p-value
  - geneset enrichment analyses
  - make it compitable with `.gmt` format
  
- Batch effect correction with 2 options
  - combaT
  - add it to design matrix (probably later versions)


## Author

- [E Onur Karakaslar](https://eonurk.github.io)

## License

- GNU General Public License v3.0

## References

- Robinson MD, McCarthy DJ, Smyth GK (2010). “edgeR: a Bioconductor package for differential expression analysis of digital gene expression data.” Bioinformatics, 26(1), 139-140. doi: 10.1093/bioinformatics/btp616.

- Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK (2015). “limma powers differential expression analyses for RNA-sequencing and microarray studies.” Nucleic Acids Research, 43(7), e47.
- Love, M.I., Huber, W., Anders, S. (2014) Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology, 15:550. 10.1186/s13059-014-0550-8



