<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to cinaR • cinaR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to cinaR">
<meta property="og:description" content="cinaR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cinaR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/cinaR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/eonurk/cinaR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="cinaR_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to cinaR</h1>
                        <h4 class="author">E Onur Karakaslar</h4>
            
            <h4 class="date">10/22/2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/eonurk/cinaR/blob/master/vignettes/cinaR.Rmd"><code>vignettes/cinaR.Rmd</code></a></small>
      <div class="hidden name"><code>cinaR.Rmd</code></div>

    </div>

    
    
<div id="quick-start" class="section level2">
<h2 class="hasAnchor">
<a href="#quick-start" class="anchor"></a>Quick Start</h2>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/eonurk/cinaR/">cinaR</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"atac_seq_consensus_bm"</span><span class="op">)</span></pre></div>
<p>Bed formatted consensus matrix (chr, start, end and samples)</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">bed</span><span class="op">)</span></pre></div>
<pre><code>## [1] 1000   25</code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="co"># bed formatted file</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">bed</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></pre></div>
<pre><code>##         Chr     Start       End B6-18mo-M-BM-47-GT18-01783
## 52834  chr5  24841478  24845196                       1592
## 29780 chr17   8162955   8164380                        109
## 67290  chr8  40577584  40578029                         72
## 51295  chr4 145277698 145278483                        110
## 4267   chr1 180808752 180815472                       2452
## 45102  chr3  88732151  88732652                         49</code></pre>
<p>Create the contrasts you want to compare, here we create contrasts for 22 mice samples from different strains.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co"># create contrast vector which will be compared.</span>
<span class="va">contrasts</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, 
              <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span><span class="op">)</span></pre></div>
<p><code>cinaR</code> function directly computes the differentially accessible peaks.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="co"># If reference genome is not set hg38 will be used!</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">bed</span>, <span class="va">contrasts</span>, reference.genome <span class="op">=</span> <span class="st">"mm10"</span><span class="op">)</span></pre></div>
<pre><code>## &gt;&gt; preparing features information...      2020-10-22 11:42:47 
## &gt;&gt; identifying nearest features...        2020-10-22 11:42:49 
## &gt;&gt; calculating distance from peak to TSS...   2020-10-22 11:42:49 
## &gt;&gt; assigning genomic annotation...        2020-10-22 11:42:49 
## &gt;&gt; assigning chromosome lengths           2020-10-22 11:43:14 
## &gt;&gt; done...                    2020-10-22 11:43:14 
## &gt;&gt; Method: edgeR
##  FDR: 0.05 &amp; abs(logFC)&lt; 0 
## &gt;&gt; Estimating dispersion...
## &gt;&gt; Fitting GLM...
## &gt;&gt; DA peaks are found!
## &gt;&gt; Human gene symbols are converted to mice!
## &gt;&gt; Enrichment results are ready...
## &gt;&gt; Done!</code></pre>
<p>Now, you can access differential accessibility (DA) and enrichment results.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></pre></div>
<pre><code>## [1] "DA.results"         "Enrichment.Results"</code></pre>
<p>Inside <code>DA.results</code>, you have the consensus peaks (cp) and differentially accessible (DA) peaks. If batch correction was run, then <code>cp</code> will be a batch-corrected consensus matrix, otherwise it is the original one you provided.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">DA.results</span><span class="op">)</span></pre></div>
<pre><code>## [1] "cp"       "DA.peaks"</code></pre>
<p>There are many information <code>cinaR</code> provides such as adjusted p value, log fold-changes, gene names etc for each peak:</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">DA.results</span><span class="op">$</span><span class="va">DA.peaks</span><span class="op">$</span><span class="va">B6_NZO</span><span class="op">)</span></pre></div>
<pre><code>##  [1] "Row.names"     "seqnames"      "start"         "end"          
##  [5] "width"         "strand"        "annotation"    "geneChr"      
##  [9] "geneStart"     "geneEnd"       "geneLength"    "geneStrand"   
## [13] "geneId"        "transcriptId"  "distanceToTSS" "gene_name"    
## [17] "logFC"         "FDR"</code></pre>
<p>Here is an overview of those DA peaks:</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">DA.results</span><span class="op">$</span><span class="va">DA.peaks</span><span class="op">$</span><span class="va">B6_NZO</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></pre></div>
<pre><code>##                  Row.names seqnames     start       end width
## 1 chr1_134559439_134560787     chr1 134559439 134560787  1349
## 2 chr1_138158514_138159483     chr1 138158514 138159483   970
## 3 chr1_164247654_164251852     chr1 164247654 164251852  4199
## 4 chr1_171631196_171631780     chr1 171631196 171631780   585
## 5 chr1_173954537_173955745     chr1 173954537 173955745  1209
## 6 chr1_177935969_177936852     chr1 177935969 177936852   884</code></pre>
<blockquote>
<p>Since the comparison is <code>B6_NZO</code>, if fold-changes are positive it means they are more accesible in B6 compared to NZO and vice versa for negative values!</p>
</blockquote>
<p>and here is a little overview for enrichment analyses results:</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">Enrichment.Results</span><span class="op">$</span><span class="va">B6_NZO</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"module.name"</span>,<span class="st">"overlapping.genes"</span>, <span class="st">"adj.p"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></pre></div>
<pre><code>##                 module.name                   overlapping.genes      adj.p
## 1         Myeloid lineage 1 PLXNC1,GM2A,AGTPBP1,CTSB,TFEB,FBXL5 0.06455975
## 2  U_metabolism/replication             PECAM1,GM2A,CTSB,SLC2A6 0.06455975
## 3  U_mitochondrial proteins     PTPRC,MAP4K4,PIK3R1,PAQR3,UBE3A 0.35904401
## 4 U_proteasome/ubiquitin cx                  PTPRC,PIK3R1,IREB2 0.38443203
## 5   U_cAMP/NF-KB activation              RUNX1,RUNX1,ETS2,CCNG2 0.38443203
## 6   U_Immunity/cytoskeleton                          RPS6,RPS19 0.57910579</code></pre>
<div id="pca-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#pca-plots" class="anchor"></a>PCA Plots</h3>
<p>You can easily get the PCA plots of the samples:</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="fu"><a href="../reference/pca_plot.html">pca_plot</a></span><span class="op">(</span><span class="va">results</span>, <span class="va">contrasts</span>, show.names <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></pre></div>
<p><img src="cinaR_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>You can overlay different information onto PCA plots as well!</p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="co"># Overlaid information</span>
<span class="va">overlaid.info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"B6-18mo"</span>, <span class="st">"B6-18mo"</span>, <span class="st">"B6-18mo"</span>, <span class="st">"B6-18mo"</span>, <span class="st">"B6-18mo"</span>, 
                   <span class="st">"NZO-18mo"</span>, <span class="st">"NZO-18mo"</span>, <span class="st">"NZO-18mo"</span>, <span class="st">"NZO-18mo"</span>, <span class="st">"NZO-18mo"</span>, <span class="st">"NZO-18mo"</span>, 
                   <span class="st">"B6-3mo"</span>, <span class="st">"B6-3mo"</span>, <span class="st">"B6-3mo"</span>, <span class="st">"B6-3mo"</span>, <span class="st">"B6-3mo"</span>, 
                   <span class="st">"NZO-3mo"</span>, <span class="st">"NZO-3mo"</span>, <span class="st">"NZO-3mo"</span>, <span class="st">"NZO-3mo"</span>, <span class="st">"NZO-3mo"</span>, <span class="st">"NZO-3mo"</span><span class="op">)</span>
<span class="co"># Sample IDs</span>
<span class="va">sample.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"S01783"</span>, <span class="st">"S01780"</span>, <span class="st">"S01781"</span>, <span class="st">"S01778"</span>, <span class="st">"S01779"</span>, 
<span class="st">"S03804"</span>, <span class="st">"S03805"</span>, <span class="st">"S03806"</span>, <span class="st">"S03807"</span>, <span class="st">"S03808"</span>, 
<span class="st">"S03809"</span>, <span class="st">"S04678"</span>, <span class="st">"S04679"</span>, <span class="st">"S04680"</span>, <span class="st">"S04681"</span>, 
<span class="st">"S04682"</span>, <span class="st">"S10918"</span>, <span class="st">"S10916"</span>, <span class="st">"S10919"</span>, <span class="st">"S10921"</span>, 
<span class="st">"S10917"</span>, <span class="st">"S10920"</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="fu"><a href="../reference/pca_plot.html">pca_plot</a></span><span class="op">(</span><span class="va">results</span>, <span class="va">overlaid.info</span>, <span class="va">sample.names</span><span class="op">)</span></pre></div>
<p><img src="cinaR_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div id="heatmaps" class="section level3">
<h3 class="hasAnchor">
<a href="#heatmaps" class="anchor"></a>Heatmaps</h3>
<p>You can also plot most variable 100 peaks for all samples:</p>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="fu"><a href="../reference/heatmap_plot.html">heatmap_plot</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></pre></div>
<p><img src="cinaR_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>Plus, you can set the number of peaks to be used in these plots, and also you can change the additional arguments of <code>pheatmap</code> function. For more information check out <code>?pheatmap</code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="fu"><a href="../reference/heatmap_plot.html">heatmap_plot</a></span><span class="op">(</span><span class="va">results</span>, heatmap.peak.count <span class="op">=</span> <span class="fl">200</span>, cluster_cols <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></pre></div>
<p><img src="cinaR_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
<div id="enrichment-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#enrichment-plots" class="anchor"></a>Enrichment Plots</h3>
<p>You can plot your enrichment results using:</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="fu"><a href="../reference/dot_plot.html">dot_plot</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></pre></div>
<pre><code>## Warning: Removed 54 rows containing missing values (geom_point).</code></pre>
<p><img src="cinaR_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<p>if it gets too crowded, you can get rid of the irrelevant pathways as well:</p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="fu"><a href="../reference/dot_plot.html">dot_plot</a></span><span class="op">(</span><span class="va">results</span>, filter.pathways <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></pre></div>
<p><img src="cinaR_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
</div>
</div>
<div id="creating-different-contrasts" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-different-contrasts" class="anchor"></a>Creating different contrasts</h2>
<p>Note that you can further divide the resolution of contrasts, for instance this is also a valid vector</p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="va">contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">bed</span><span class="op">)</span>, split <span class="op">=</span> <span class="st">"-"</span>, fixed <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>, 
                    <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>, collapse <span class="op">=</span> <span class="st">"-"</span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">25</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">contrasts</span><span class="op">)</span></pre></div>
<pre><code>## [1] "B6-18mo-M-BM"  "B6-18mo-F-BM"  "NZO-18mo-F-BM" "NZO-18mo-M-BM"
## [5] "B6-3mo-F-BM"   "B6-3mo-M-BM"   "NZO-3mo-F-BM"  "NZO-3mo-M-BM"</code></pre>
<p>in this case, each of them will be compared to each other which will result in 28 different comparisons.</p>
</div>
<div id="running-enrichment-with-different-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#running-enrichment-with-different-dataset" class="anchor"></a>Running enrichment with different dataset</h2>
<p>You can run the enrichment analyses with a custom geneset:</p>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="fu"><a href="../reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">...</span>, geneset <span class="op">=</span> <span class="va">new_geneset</span><span class="op">)</span></pre></div>
<p><code>geneset</code> must be a <code>.gmt</code> formatted symbol file. You can download different genesets from this <a href="https://www.gsea-msigdb.org/gsea/downloads.jsp">site</a>.</p>
<blockquote>
<p>You can use <code>read.gmt</code> function from <code>qusage</code> package to read genesets into your current environment.</p>
</blockquote>
<p>Also, you can familarize yourself with the format by checking out :</p>
<div class="sourceCode" id="cb30"><pre class="downlit">
<span class="co"># default geneset to be used</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"VP2008"</span><span class="op">)</span></pre></div>
</div>
<div id="selecting-different-reference-genomes" class="section level2">
<h2 class="hasAnchor">
<a href="#selecting-different-reference-genomes" class="anchor"></a>Selecting different reference genomes</h2>
<p>For now, <code>cinaR</code> supports 3 genomes for human and mice models:</p>
<ul>
<li><code>hg38</code></li>
<li><code>hg19</code></li>
<li><code>mm10</code></li>
</ul>
<p>You can set your it using <code>reference.genome</code> argument.</p>
</div>
<div id="batch-effect-correction" class="section level2">
<h2 class="hasAnchor">
<a href="#batch-effect-correction" class="anchor"></a>Batch Effect Correction</h2>
<p>If you suspect your data have unknown batch effects, you can use:</p>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="fu"><a href="../reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">...</span>, batch.correction <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></pre></div>
<p>This option will run <a href="http://bioconductor.org/packages/release/bioc/html/sva.html">Surrogate Variable Analysis</a> (SVA) and try to adjust your data for unknown batch effects. If however, you already know the batches of the samples, you can simply set the <code>batch.information</code> argument as well:</p>
<div class="sourceCode" id="cb32"><pre class="downlit">
<span class="co"># batch information should be number a vector where</span>
<span class="co"># the length of it equals to the number of samples.</span>
<span class="fu"><a href="../reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">...</span>, batch.correction <span class="op">=</span> <span class="cn">T</span>, batch.information <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">11</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">11</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></pre></div>
<blockquote>
<p>Reminder - In our example data we have 22 samples</p>
</blockquote>
</div>
<div id="saving-da-peaks-to-excel" class="section level2">
<h2 class="hasAnchor">
<a href="#saving-da-peaks-to-excel" class="anchor"></a>Saving DA peaks to excel</h2>
<p>Setting <code>save.DA.peaks = TRUE</code> in <code>cinaR</code> function will create a <code>DApeaks.xlsx</code> file in the current directory. This file includes all the comparisons in different tabs. Additionally, you can set the path/name of the file using <code>DA.peaks.path</code> argument after setting <code>save.DA.peaks = TRUE</code>.</p>
<p>For instance,</p>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">...</span>, save.DA.peaks <span class="op">=</span> <span class="cn">T</span>, DA.peaks.path <span class="op">=</span> <span class="st">"./Peaks_mice.xlsx"</span><span class="op">)</span></pre></div>
<p>will create an excel file with name <code>Peaks_mice.xlsx</code> in the current directory.</p>
</div>
<div id="using-different-glm-algorithms" class="section level2">
<h2 class="hasAnchor">
<a href="#using-different-glm-algorithms" class="anchor"></a>Using different GLM algorithms</h2>
<p>Currently, <code>cinaR</code> supports 4 different algorithms, namely;</p>
<ul>
<li>edgeR</li>
<li>limma-voom</li>
<li>limma-trend</li>
<li>DESeq2</li>
</ul>
<p>If not set, it uses <code>edgeR</code> for differential analyses. You can change the used algorithm by simply setting <code>DA.choice</code> argument. For more information, <code><a href="../reference/cinaR.html">?cinaR</a></code></p>
</div>
<div id="some-useful-arguments" class="section level2">
<h2 class="hasAnchor">
<a href="#some-useful-arguments" class="anchor"></a>Some useful arguments</h2>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="co"># new FDR threshold for DA peaks</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">...</span>, DA.fdr.threshold <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>

<span class="co"># filters out pathways</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">...</span>, enrichment.FDR.cutoff <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span>

<span class="co"># does not run enrichment pipeline</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">...</span>, run.enrichment <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co"># creates the piechart from chIpSeeker package</span>
<span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">...</span>, show.annotation.pie <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># change cut-off value for dot plots</span>
<span class="fu"><a href="../reference/dot_plot.html">dot_plot</a></span><span class="op">(</span><span class="va">...</span>, fdr.cutoff <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></pre></div>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<ul>
<li><p>Robinson MD, McCarthy DJ, Smyth GK (2010). “edgeR: a Bioconductor package for differential expression analysis of digital gene expression data.” Bioinformatics, 26(1), 139-140. doi: 10.1093/bioinformatics/btp616.</p></li>
<li><p>Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK (2015). “limma powers differential expression analyses for RNA-sequencing and microarray studies.” Nucleic Acids Research, 43(7), e47.</p></li>
<li><p>Love, M.I., Huber, W., Anders, S. (2014) Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology, 15:550. 10.1186/s13059-014-0550-8</p></li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Onur Karakaslar, Duygu Ucar.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
